app/models/manageiq/providers/amazon/storage_manager/s3/cloud_object_store_container.rb

```diff
@@ -23,7 +23,7 @@ class ManageIQ::Providers::Amazon::StorageManager::S3::CloudObjectStoreContainer
   end
 
   def provider_object(connection = nil)
-    connection ||= ext_management_system.connect
+    connection = ext_management_system.connect_regional(provider_region)
```

Comment:
The ||= should stay, because it allows the caller to override the connection used by passing one in (pretty sure this was a dependency injection for testing purposes).